lib/events.js

```diff
@@ -264,9 +271,102 @@ ObjectDefineProperty(EventEmitter.prototype, kCapture, {
   enumerable: false,
 });
 
-EventEmitter.prototype._events = undefined;
-EventEmitter.prototype._eventsCount = 0;
+function isEventUnsupportedForFastPath(type) {
+  // Not supporting newListener and removeListener events in fast path
+  // as they can add new listeners in the middle of the process
+  // and also not supporting errorMonitor event as it has special handling
+  return type === 'newListener' || type === 'removeListener' || type === kErrorMonitor;
+}
+
+ObjectDefineProperties(EventEmitter.prototype, {
+  [kImpl]: {
+    __proto__: null,
+    value: undefined,
+    enumerable: false,
+    configurable: false,
+    writable: true,
+  },
+  [kIsFastPath]: {
+    __proto__: null,
+    value: undefined,
+    enumerable: false,
+    configurable: false,
+    writable: true,
+  },
+  [kInitialFastPath]: {
+    __proto__: null,
+    value: undefined,
+    enumerable: false,
+    configurable: false,
+    writable: true,
+  },
+  _events: {
+    __proto__: null,
+    enumerable: true,
+
+    get: function() {
+      // TODO - remove optional chaining
+      return this[kImpl]?._events;
+    },
+    set: function(arg) {
+
+      if (this[kImpl] === undefined && this[kInitialFastPath] === true) {
+        // TODO(rluvaton): remove this eslint ignore
+        // eslint-disable-next-line no-use-before-define
+        this[kImpl] = new FastEventEmitter(this, arg);
+        this[kIsFastPath] = true;
+        this[kInitialFastPath] = undefined;
+        return;
+      }
+
+      if (this[kImpl] === undefined) {
+
+        // TODO(rluvaton): find a better way and also support array with length 1?
+        // TODO(rluvaton): if have newListener and removeListener events, switch to slow path
+
+        const shouldBeFastPath = arg === undefined ||
+          ObjectEntries(arg).some(({ 0: key, 1: value }) =>
```

Comment:
Can you avoid `Object.entries` in here?

Reply:
Instead changed to a slow path by default to avoid potential bugs